archunit/src/main/java/com/tngtech/archunit/core/domain/JavaCodeUnit.java

```diff
@@ -43,10 +44,11 @@
  * in particular every place, where Java code with behavior, like calling other methods or accessing fields, can
  * be defined.
  */
-public abstract class JavaCodeUnit extends JavaMember implements HasParameterTypes, HasReturnType {
+public abstract class JavaCodeUnit extends JavaMember implements HasParameterTypes, HasReturnType, HasThrowsTypes {
     private final JavaClass returnType;
     private final List<JavaClass> parameters;
     private final String fullName;
+    private final List<JavaClass> throwsTypes;
```

Comment:
I would call this `throwsDeclarations` and I would not make this of type `List<JavaClass>`, but `List<ThrowsDeclaration>`. A ThrowsDeclaration can offer `JavaClass ThrowsDeclaration.getType()`, but I don't think it should be modeled as a `JavaClass` (which it semantically is not).
I made the same mistake with `parameters` being of type `List<JavaClass>`, now I want to introduce parameter annotations and it's hardly possible without breaking the API :frowning_face: 
The reason in the end is always the same though, type mismatch. A parameter is not really a `JavaClass` and neither is a throws declaration.